File: clients/java/src/main/java/model/Versioned.java
package model;

import util.StreamUtil;

public class Versioned {
    private java.util.Map<Integer, model.UnitAction> inner;
    public java.util.Map<Integer, model.UnitAction> getInner() { return inner; }
    public void setInner(java.util.Map<Integer, model.UnitAction> inner) { this.inner = inner; }
    public Versioned() {}
    public Versioned(java.util.Map<Integer, model.UnitAction> inner) {
        this.inner = inner;
    }
    public static Versioned readFrom(java.io.InputStream stream) throws java.io.IOException {
        Versioned result = new Versioned();
        int innerSize = StreamUtil.readInt(stream);
        result.inner = new java.util.HashMap<>(innerSize);
        for (int i = 0; i < innerSize; i++) {
            int innerKey;
            innerKey = StreamUtil.readInt(stream);
            model.UnitAction innerValue;
            innerValue = model.UnitAction.readFrom(stream);
            result.inner.put(innerKey, innerValue);
        }
        return result;
    }
    public void writeTo(java.io.OutputStream stream) throws java.io.IOException {
        StreamUtil.writeInt(stream, 43981);
        StreamUtil.writeInt(stream, inner.size());
        for (java.util.Map.Entry<Integer, model.UnitAction> innerEntry : inner.entrySet()) {
            int innerKey = innerEntry.getKey();
            model.UnitAction innerValue = innerEntry.getValue();
            StreamUtil.writeInt(stream, innerKey);
            innerValue.writeTo(stream);
        }
    }
}

File: clients/javascript/model/versioned.js
const UnitAction = require('./unit-action').UnitAction;
class Versioned {
    constructor(inner) {
        this.inner = inner;
    }
    static async readFrom(stream) {
        let inner;
        inner = new Map();
        for (let i = await stream.readInt(); i > 0; i--) {
            let innerKey;
            let innerValue;
            innerKey = await stream.readInt();
            innerValue = await UnitAction.readFrom(stream);
            inner.set(innerKey, innerValue);
        }
        return new Versioned(inner);
    }
    async writeTo(stream) {
        await stream.writeInt(43981);
        let inner = this.inner;
        await stream.writeInt(inner.size);
        for (let innerEntry of inner) {
            let innerKey = innerEntry[0];
            let innerValue = innerEntry[1];
            await stream.writeInt(innerKey);
            await innerValue.writeTo(stream);
        }
    }
}
module.exports = { Versioned: Versioned }
